### ISSUE TITLE
Design custom select styles for users navigating by keyboard

### ISSUE THREAD

**Issue by gissoo** [Princeton-CDH/startwords#71]

**Comment by gissoo** [comment]
@rlskoeser @thatbudakguy I have created select styles for keyboard navigation. Do you have any thoughts on this?

**Comment by rlskoeser** [comment]
They seem nice, I just wonder if they are visible enough. Did you check, do these color combinations provide good contrast?

Keyboard nav is often used by people with reduced eyesight who need more help identifying where they are. If we're customizing the select style we want it to be at least as visible as the browser default.

**Comment by gissoo** [comment]
@rlskoeser Thanks for writing! This is a bit tricky to test – also very challenging to write here about it now – I think it's best we briefly converse on it next week – I'll share another alternative before we converse though.

**Comment by thatbudakguy** [comment]
not sure if the figma link is working for me...@gissoo can you describe what the styles are? I can't see anything different on the page.

**Comment by gissoo** [comment]
@rlskoeser @thatbudakguy What I had initially designed, labeled as "Navigation by Keyboard – V1" on the page, needed multiple color tests for accessibility and some of them failed it's out of the picture for me, I have two more versions labeled as "Navigation by Keyboard – V2", and "Navigation by Keyboard – V3". Please let me know what you think, and here, there will have to be two colors to ensure visibility on the light and dark backgrounds. [Here are the designs](https://www.figma.com/file/MEt3gfKX4N3WTLPbLGBbGC/Startwords?node-id=688%3A1772)

@thatbudakguy If you can't still can't find the designs on this page we can zoom and we can share screens so I can understand better what might be happening.

**Comment by thatbudakguy** [comment]
@gissoo I was able to find it, thanks for the link!

I think this version is probably quite easy to implement since it's just a simple background color, so that's nice. I'm still not sure it has enough contrast, though.

For the pink background on white (e.g. on the issue list page), figma shows the color as `#ce2949` at `0.1` (10%) opacity. Using [this converter](https://borderleft.com/toolbox/rgba/), that color overlaid on white is effectively `#faeaed`. According to [webAIM's contrast checker](https://webaim.org/resources/contrastchecker/), `#faeaed` and white `#ffffff` have a contrast ratio of 1.16:1, which fails the WCAG AA standard of 3:1 for graphics and UI components (like form input borders).

**Comment by rlskoeser** [comment]
In general, I like the approach in V1 better than V2 and V3 — the boxy outlines seem like they don't fit that well with the existing site visuals. But of course contrast is important!

@thatbudakguy are the different background colors in V1 easy to distinguish based on the context of the links or classes already applied? What do you think of V2 and V3 ? 

@gissoo I don't know what some of the arrows on "Article - mobile - keyboard" in V1 are pointing to, particularly around the footnote reference and contextual note. Is there an error here with extra arrows and an extra footnote reference number?

**Comment by gissoo** [comment]
@thatbudakguy @rlskoeser Thanks for writing. So as I mentioned v1 fails a lot of the color contrast requirements for accessibility when I tested again yesterday (there are two sets of color requirements I had to follow yesterday one for the box on the page background and then another one for the text on the box :D ) and I think this feature especially should not be failing accessibility and should only support it. I can try the background style but with different colors (more visible) and see how it works. 

@rlskoeser The arrows were just placed so you see all the places I have thought of and applied the styles for keyboard nav, particularly on the article page. There are two article pages for v2 and v3 because I wanted to try two styles for the "x" button on the contextual container. And yes, the extra footnote ref number was a mistake, thanks for catching, it's deleted.

**Comment by thatbudakguy** [comment]
sorry, was just now able to find V2 and V3. the screen is so big!!

I actually like both V2 and V3 better than V1, and I like V3 the best. The outline is a predictable pattern for focus highlights, and the dotted outline is something I think I've seen in a lot of accessible examples from the W3C and deque.

**Comment by rlskoeser** [comment]
I defer to @thatbudakguy here. I do like V3 better than V2. 

It seems like in some places (DOI, pdf, text links) we need more padding for the text to be legible with the dotted outline visible.

@gissoo is the focus style for the contextual note reference link the same as the hover? (just asking for clarification, this seems fine to me)

I wish we could do a customized shape for the outline that matches the rest of the site (e.g. just notching two corners like the top left corner nav), but I don't know if that actually makes any logical sense and probably turns what is a very simple style implementation into something much more complicated and likely to cause problems.

**Comment by gissoo** [comment]
@thatbudakguy @rlskoeser Thanks for writing, @thatbudakguy that's fine!! I tested further with the v1 style but I don't think we can make it work (Too many color restrictions) – I agree with v3 too! I have revised it to address the issues @rlskoeser brought up. Also, thank you @rlskoeser for pointing out the contextual note reference link (I had not updated it) – it's updated now. It will have the hover style and the black border, I think it will be more helpful to have both. [Here are the designs](https://www.figma.com/file/MEt3gfKX4N3WTLPbLGBbGC/Startwords?node-id=688%3A1772)

@rlskoeser I'm hesitant to do a customized shape for this feature. It seems too much to me, it cam diminish/take away from the uniqueness of the site's visual style if it's in too many places.

**Comment by rlskoeser** [comment]
@gissoo thanks for explaining. I'm happy with these.

**Comment by gissoo** [comment]
@gwijthoff Please take a look [the keyboard navigation styles (v3) I have designed here](https://www.figma.com/file/MEt3gfKX4N3WTLPbLGBbGC/Startwords?node-id=688%3A1772) and let me know what you think and if you any thoughts or concerns.

**Comment by gwijthoff** [comment]
Looks great, @gissoo — thanks for your work on this! Especially love the circular outline around the X to close out contextual notes.

**Comment by gissoo** [comment]
@gwijthoff thanks for writing! Please close the issue if no issues then.
